python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+
+        if max_magNorm_err < 0.001:
+            print('Pass: Image MagNorms are within 0.001 of correct values.')
+        else:
+            raise CatalogError('\nFail: Max image phosim MagNorm error is greater than 0.001 mags. ' + 
+                               'Max error is: %.4f mags.' % max_magNorm_err)
+
+        print('------------')
+
+        return
+
+    def get_agn_variability_mags(self, var_param_dict, time_delay, mjd, redshift):
+
+        def return_num_obj(params):
+            return 1
+
+        eg_test = egvar()
+        eg_test.num_variable_obj = return_num_obj
+
+        var_mags = eg_test.applyAgn([[0]], var_param_dict, 
+                                    mjd-time_delay,
+                                    redshift=np.array([redshift]))
+
+        filters = ['u', 'g', 'r', 'i', 'z', 'y']
+        var_mag_dict = {x:y for x,y in zip(filters, var_mags)}
+
+        return var_mag_dict
+
+        
+    def load_agn_var_params(self):
+
+        agn_var_params = {}
+
+        with open(os.path.join(os.environ['TWINKLES_DIR'], 'data', 
+                                          'agn_validation_params.txt'), 'r') as f:
+            for line in f:
+                line_id = line.split(',')[0]
+                line_magNorm = line.split(',')[3]
+                
+                var_param_dict = {x.split(':')[0][2:-1]:np.array([np.float(x.split(':')[1])]) 
+                                  for x in line.split('{')[2][:-3].split(',')}
+
+                # Some munging from the way we made the dict
+                var_param_dict['seed'] = np.array(var_param_dict['eed'], dtype=int)
+                var_param_dict['magNorm_static'] = np.float(line_magNorm)
+                del var_param_dict['eed']
+
+                agn_var_params[str(line_id)] = var_param_dict
+
+        return agn_var_params
+
+    def compare_sne_image_mags(self, spr_sne_df, spr_sne_lens_df, visit_mjd,
+                               visit_band):
+
+        """
+        This test compares the image magnitudes.
+        """
+        
+        sn_obj = SNObject(0., 0.)
+
+        df = pd.read_csv(os.path.join(os.environ['TWINKLES_DIR'], 'data',
+                                      'dc2_sne_cat.csv'))
+
+        bandpassDict = BandpassDict.loadTotalBandpassesFromFiles(bandpassNames=[visit_band])
+        norm_bp = Bandpass()
+        norm_bp.imsimBandpass()
+
+        max_magNorm_err = 0.
+
+        for lens_gal_row in spr_sne_lens_df.iterrows():
+
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            spr_sys_df = spr_sne_df.query('lens_galaxy_uID == %i' % u_id)
+
+            # There may be no images present in the current instance catalog
+            if len(spr_sys_df) == 0:
+                continue
+
+            lens = df.query('twinkles_sysno == %i' % spr_sys_df['twinkles_system'].iloc[0])
+            img_vals = spr_sys_df['image_number'].values
+
+            mag = lens['mu']
```

Comment:
Given that you take the `log` of this below, I gather that this is a flux.  It is probably best to call it `flux`.

Reply:
Actually a magnification factor on the flux. Therefore I will label it `magnification` to be clearer.